docs/advanced.rst

```diff
@@ -1018,105 +1136,665 @@ to learn more about the use of this feature.
+| **XML**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-DISCOVERY-PROTOCOL<-->   |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+.. _DS_guidPrefx:
+
+The server unique identifier ``GuidPrefix``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This belongs to the RTPS specification and univocally identifies each DDS participant. It consist on 12 bytes and is
+basically a key in the DDS domain.
+In the server-client discovery has the purpose to link a *server* to its *clients*.
+Note that there is an auxiliary **ReadguidPrefix** method to populate the ``GuidPrefix`` using a ``string``.
+It must be mandatorily specified in:
+
+Server side setup
+"""""""""""""""""
+
+In the examples below we show how to setup the attribute:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.prefix
+
+and xml tag:
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>prefix
+
++------------------------------------------------------------+
+| **C++**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp                 |
+|    :language: c++                                          |
+|    :start-after: //CONF_SERVER_SERVER_GUIDPREFIX           |
+|    :end-before: //!--                                      |
++------------------------------------------------------------+
+| **XML**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-SERVER-PREFIX<-->        |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+Note that a *server* can act as a *client* of other *servers*.
+Thus, the following section may also apply.
+
+Client side setup
+""""""""""""""""""
+
+Each *client* must keep a list of the *servers* it wants to link to.
+Each single element represents an individual server and a ``GuidPrefix`` must be provided.
+The *server* list is the attribute:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.builtin.discovery_config.m_DiscoveryServers
+
+and must be populated with ``RemoteServerAttributes`` objects with a valid ``guidPrefix`` member.
+In xml the server list is and its elements are simultaneously specified.
+Note that ``prefix`` is an attribute of the ``RemoteServer`` tag.
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>builtin>discovery_config>discoveryServerList>RemoteServer@prefix
```

Comment:
Get rid of the enum and tag hierarchy, as they are implicit on the examples.

Reply:
As above... It's not easy to imply the jigsaw puzzle that the attributes are, directly, from the examples...that's the main reason I put these clues there.

Reply:
👍